core-patch/src/main/java/eu/neverblink/jelly/core/patch/JellyPatchOptions.java

```diff
@@ -0,0 +1,158 @@
+package eu.neverblink.jelly.core.patch;
+
+import static eu.neverblink.jelly.core.JellyOptions.checkTableSize;
+
+import eu.neverblink.jelly.core.JellyOptions;
+import eu.neverblink.jelly.core.RdfProtoDeserializationError;
+import eu.neverblink.jelly.core.proto.v1.PatchStatementType;
+import eu.neverblink.jelly.core.proto.v1.PatchStreamType;
+import eu.neverblink.jelly.core.proto.v1.PhysicalStreamType;
+import eu.neverblink.jelly.core.proto.v1.RdfPatchOptions;
+import eu.neverblink.jelly.core.proto.v1.RdfStreamOptions;
+
+/**
+ * Utilities for working with RdfPatchOptions.
+ */
+public class JellyPatchOptions {
+
+    private JellyPatchOptions() {}
+
+    public static final RdfPatchOptions BIG_STRICT = fromJellyOptions(JellyOptions.BIG_STRICT);
+
+    public static final RdfPatchOptions BIG_GENERALIZED = fromJellyOptions(JellyOptions.BIG_GENERALIZED);
+
+    public static final RdfPatchOptions BIG_RDF_STAR = fromJellyOptions(JellyOptions.BIG_RDF_STAR);
+
+    public static final RdfPatchOptions BIG_ALL_FEATURES = fromJellyOptions(JellyOptions.BIG_ALL_FEATURES);
+
+    public static final RdfPatchOptions SMALL_STRICT = fromJellyOptions(JellyOptions.SMALL_STRICT);
+
+    public static final RdfPatchOptions SMALL_GENERALIZED = fromJellyOptions(JellyOptions.SMALL_GENERALIZED);
+
+    public static final RdfPatchOptions SMALL_RDF_STAR = fromJellyOptions(JellyOptions.SMALL_RDF_STAR);
+
+    public static final RdfPatchOptions SMALL_ALL_FEATURES = fromJellyOptions(JellyOptions.SMALL_ALL_FEATURES);
+
+    public static final RdfPatchOptions DEFAULT_SUPPORTED_OPTIONS = fromJellyOptions(
+        JellyOptions.DEFAULT_SUPPORTED_OPTIONS
+    );
+
+    /**
+     * Convert a Jelly RdfStreamOptions to a Jelly Patch RdfPatchOptions.
+     * <p>
+     * GRAPHS physical type is converted to QUADS. Logical stream type and other fields that are not
+     * relevant to RDF Patch are ignored.
+     *
+     * @param opt RdfStreamOptions
+     * @return RdfPatchOptions
+     */
+    public static RdfPatchOptions fromJellyOptions(RdfStreamOptions opt) {
+        return fromBaseOptions(opt).toBuilder().setStatementType(fromJellyPhysicalType(opt.getPhysicalType())).build();
+    }
+
+    /**
+     * Convert a BaseJellyOptions instance to a Jelly Patch RdfPatchOptions.
+     * @param opt BaseJellyOptions
+     * @return RdfPatchOptions
+     */
+    public static RdfPatchOptions fromBaseOptions(RdfStreamOptions opt) {
```

Comment:
Ugh, this got messy.

BaseJellyOptions was useful because we had a single iface for both RdfStream and RdfPatch options, so we had no code duplication in the validation methods (checkCompatibility).

We will replace the proto classes anyway before shipping 3.0.0, so this *can* stay like this for now. But please open an issue to untangle this mess and restore the old class hierarchy once we gain the ability to force the proto classes to extend the same interface.